neon-annotations/src/main/java/com/zing/zalo/neon2/annoations/NeonObject.java

```diff
@@ -0,0 +1,27 @@
+package com.zing.zalo.neon2.annoations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.TYPE)
+public @interface NeonObject {
+    int version() default 0;
+
+    int compatibleSince() default 0;
+
+    boolean inheritanceSupported() default true;
+
+    @Target(ElementType.FIELD) @interface Property {
+        int sinceVersion() default 0;
+
+        @Deprecated
+        int arraySize() default 0;
```

Comment:
@tienlocbui1110 I think we don't need this if we move the core to twitter/Serial

Reply:
I think it too. We just need to support inheritance field. But version is supported by default.